images/macos/provision/core/homebrew.sh

```diff
@@ -1,23 +1,39 @@
-#!/bin/sh
-echo "Installing Homebrew..."
+#!/bin/bash -e -o pipefail
 
 source ~/utils/utils.sh
 
-echo Installing Homebrew...
-HOMEBREW_INSTALL_URL="https://raw.githubusercontent.com/Homebrew/install/master/install.sh"
+echo "Installing Command Line Tools..."
+install_clt
+
+# Retry the installation if tools are not installed from the first attempt
+retries=30
+sleepInterval=60
+while ! is_clt_installed; do
+    if [[ $retries -eq 0 ]]; then
+        echo "Unable to find the Command Line Tools, all the attempts exhausted"
+        exit 1
+    fi
+    echo "Command Line Tools not found, trying to install them via softwareupdates, $retries attempts left"
+    install_clt
+    ((retries--))
+    echo "Wait $sleepInterval seconds before the next check for installed Command Line Tools"
+    sleep $sleepInterval
+done
```

Comment:
could you please keep this logic in the separate file `xcode-ctl.sh`. I don't think it makes sense to mix it with homebrew installation. Actually, they are not related

Reply:
In this case, you can put both `is_clt_installed` and `install_clt` directly to `xcode-ctl.sh` and don't put them to helper file because they are not re-usable